integration/setup/provider.go

```diff
@@ -2,38 +2,328 @@ package setup
+						{
+							VMSize: "Standard_D4_v2",
+						},
+					},
+					VirtualNetwork: providerv1alpha1.AzureConfigSpecAzureVirtualNetwork{
+						CIDR:             env.AzureCIDR(),
+						MasterSubnetCIDR: env.AzureMasterSubnetCIDR(),
+						WorkerSubnetCIDR: env.AzureWorkerSubnetCIDR(),
+						CalicoSubnetCIDR: env.AzureCalicoSubnetCIDR(),
+					},
+					Workers: []providerv1alpha1.AzureConfigSpecAzureNode{
+						{
+							VMSize: "Standard_D4_v2",
+						},
+						{
+							VMSize: "Standard_D4_v2",
+						},
+					},
+				},
+				Cluster: providerv1alpha1.Cluster{
+					Calico: providerv1alpha1.ClusterCalico{
+						CIDR: 16,
+						MTU:  1500,
+					},
+					Customer: providerv1alpha1.ClusterCustomer{ID: "example-customer"},
+					Docker:   providerv1alpha1.ClusterDocker{Daemon: providerv1alpha1.ClusterDockerDaemon{CIDR: "172.17.0.1/16"}},
+					Etcd: providerv1alpha1.ClusterEtcd{
+						Domain: "etcd." + env.ClusterID() + ".k8s." + env.CommonDomain(),
+						Port:   2379,
+						Prefix: "giantswarm.io",
+					},
+					ID: env.ClusterID(),
+					Kubernetes: providerv1alpha1.ClusterKubernetes{
+						API: providerv1alpha1.ClusterKubernetesAPI{
+							ClusterIPRange: "172.31.0.0/16",
+							Domain:         "api." + env.ClusterID() + ".k8s." + env.CommonDomain(),
+							SecurePort:     443,
+						},
+						CloudProvider: "azure",
+						DNS:           providerv1alpha1.ClusterKubernetesDNS{IP: net.IPv4(172, 31, 0, 10)},
+						Domain:        "cluster.local",
+						IngressController: providerv1alpha1.ClusterKubernetesIngressController{
+							Domain:         "ingress." + env.ClusterID() + ".k8s." + env.CommonDomain(),
+							WildcardDomain: "*." + env.ClusterID() + ".k8s." + env.CommonDomain(),
+							InsecurePort:   30010,
+							SecurePort:     30011,
+						},
+						Kubelet: providerv1alpha1.ClusterKubernetesKubelet{
+							AltNames: "kubernetes,kubernetes.default,kubernetes.default.svc,kubernetes.default.svc.cluster.local",
+							Domain:   "worker." + env.ClusterID() + ".k8s." + env.CommonDomain(),
+							Labels:   "giantswarm.io/provider=azure,azure-operator.giantswarm.io/version=" + version,
+							Port:     10250,
+						},
+						NetworkSetup: providerv1alpha1.ClusterKubernetesNetworkSetup{Docker: providerv1alpha1.ClusterKubernetesNetworkSetupDocker{Image: "quay.io/giantswarm/k8s-setup-network-environment:1f4ffc52095ac368847ce3428ea99b257003d9b9"}},
+						SSH:          nodeSSHConfiguration,
+					},
+				},
+				VersionBundle: providerv1alpha1.AzureConfigSpecVersionBundle{Version: version},
+			},
+		}
+		_, err := config.K8sClients.G8sClient().ProviderV1alpha1().AzureConfigs("default").Create(azureConfig)
 		if err != nil {
 			return microerror.Mask(err)
 		}
+	}
 
-		err = config.Release.Install(ctx, "credentiald", release.NewStableVersion(), values, config.Release.Condition().SecretExists(ctx, namespace, "credential-default"))
+	{
+		azureClusterConfig := &v1alpha1.AzureClusterConfig{
```

Comment:
We need cluster-operator to react to AzureClusterConfig so it creates the required certificates. Before this PR, the certificates were being installed using e2etemplates.